task/upload.go

```diff
@@ -72,27 +74,57 @@ func handleUploadVOD(p handleUploadVODParams) (*TaskHandlerOutput, error) {
 		if err != nil {
 			return nil, err
 		}
-		var encryption *clients.EncryptionPayload
-		params := tctx.Task.Params.Upload
 
-		if params != nil && params.Encryption.EncryptedKey != "" {
-			encryption = &clients.EncryptionPayload{
-				EncryptedKey: params.Encryption.EncryptedKey,
+		var encryption *clients.EncryptionPayload
+		var clipStrategy *video.ClipStrategy
+
+		if tctx.Task.Params.Clip != nil {
+			clipParams := tctx.Task.Params.Clip
+			if clipParams != nil {
+				clipStrategy = &video.ClipStrategy{
+					StartTime:  clipParams.ClipStrategy.StartTime,
+					EndTime:    clipParams.ClipStrategy.EndTime,
+					PlaybackID: clipParams.ClipStrategy.PlaybackId,
+				}
+			}
+		} else {
+			uploadParams := tctx.Task.Params.Upload
+			if uploadParams != nil && uploadParams.Encryption.EncryptedKey != "" {
+				encryption = &clients.EncryptionPayload{
+					EncryptedKey: uploadParams.Encryption.EncryptedKey,
+				}
 			}
 		}
-		var (
-			req = clients.UploadVODRequest{
-				ExternalID:            tctx.Task.ID,
-				Url:                   inUrl,
-				CallbackUrl:           tctx.catalyst.CatalystHookURL(tctx.Task.ID, "finalize", catalystTaskAttemptID(tctx.Task)),
-				OutputLocations:       outputLocations,
-				PipelineStrategy:      p.catalystPipelineStrategy,
-				Profiles:              p.profiles,
-				TargetSegmentSizeSecs: p.targetSegmentSizeSecs,
-				Encryption:            encryption,
+
+		var req clients.UploadVODRequest
+		req = clients.UploadVODRequest{
+			ExternalID:            tctx.Task.ID,
+			Url:                   inUrl,
+			CallbackUrl:           tctx.catalyst.CatalystHookURL(tctx.Task.ID, "finalize", catalystTaskAttemptID(tctx.Task)),
+			OutputLocations:       outputLocations,
+			PipelineStrategy:      p.catalystPipelineStrategy,
+			Profiles:              p.profiles,
+			TargetSegmentSizeSecs: p.targetSegmentSizeSecs,
+			Encryption:            encryption,
+		}
+
+		if clipStrategy != nil {
+			req.ClipStrategy = clients.ClipStrategy{
+				StartTime:  clipStrategy.StartTime,
+				EndTime:    clipStrategy.EndTime,
+				PlaybackID: clipStrategy.PlaybackID,
+			}
+		} else {
+			// TODO : this is just temporary for staging - to remove
+			req.ClipStrategy = clients.ClipStrategy{
+				StartTime:  0,
+				EndTime:    0,
+				PlaybackID: "",
 			}
```

Comment:
This is not necessary. Go already initializes objects with the 0 value. I think it was a noop even in staging.

Reply:
We just needed this temporarily because I was inadvertently setting StartTime to be a required value on catalyst-api. We can remove this now.

Reply:
I only meant that this assignment here is a noop. It will not change what is the serialization of the `ClipStrategy` field since it doesn't change the value of any of its fields (and it's stored in `req` by value, not by pointer)

What would really make a difference is the `omitempty` flag on the json tag (but even then, this assignment is noop). A playground: https://go.dev/play/p/4z6mTYrk59_Y